src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentState = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
+    const action = actions[i];
+
+    if (action.type === 'clear') {
+      currentState = {};
+    } else if (action.type === 'addProperties') {
+      currentState = { ...currentState, ...action.extraData };
+    } else if (action.type === 'removeProperties') {
+      currentState = { ...currentState };
+
+      for (let j = 0; j < action.keysToRemove.length; j++) {
+        const key = action.keysToRemove[j];
+
+        delete currentState[key];
+      }
+    }
+
+    stateHistory.push(currentState);
```

Comment:
Pushing the currentState to the stateHistory array is being repeated in every condition. Consider moving this operation outside of the if-else block, after the current action is processed, to adhere to the DRY (Don't Repeat Yourself) principle.

Reply:
nevermind